include/envoy/http/codec.h

```diff
@@ -148,11 +148,25 @@ class ConnectionCallbacks {
 };
 
 /**
- * A list of options that can be specified when creating a codec.
+ * http/2 codec settings
  */
-class CodecOptions {
-public:
-  static const uint64_t NoCompression = 0x1;
+struct Http2Settings {
+  uint32_t hpack_table_size_{DEFAULT_HPACK_TABLE_SIZE};
+  uint32_t max_concurrent_streams_{DEFAULT_MAX_CONCURRENT_STREAMS};
+  uint32_t initial_window_size_{DEFAULT_INITIAL_WINDOW_SIZE};
+
+  static const uint32_t MIN_HPACK_TABLE_SIZE = 0;
+  static const uint32_t DEFAULT_HPACK_TABLE_SIZE = 4 * 1024; // from nghttp2
+  static const uint32_t MAX_HPACK_TABLE_SIZE = 16 * 1024 * 1024;
+
+  static const uint32_t MIN_MAX_CONCURRENT_STREAMS = 1;
+  static const uint32_t DEFAULT_MAX_CONCURRENT_STREAMS = 1024;
+  static const uint32_t MAX_MAX_CONCURRENT_STREAMS = 1 << 29; // from MAX_STREAMS
+
+  static const uint32_t MIN_INITIAL_WINDOW_SIZE = (1 << 16) - 1; // from HTTP/2 spec
+  static const uint32_t DEFAULT_INITIAL_WINDOW_SIZE = 256 * 1024 * 1024;
+  // ( 1 << 31 ) - 1 cause overflow warning, so hard-code the value
```

Comment:
You can avoid the overflow by specifying the 1 as an unsigned long long literal:

`(1ULL << 31) - 1`